### PR TITLE
Articles

### PR DESCRIPTION
**Pull request by TheDimDom**

### CODE REVIEW

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -43,3 +43,39 @@ describe("/api", () => {
...
+  });
+
+  test("400: checks for bad request with article_id = 0", () => {
+    return request(app)
+      .get("/api/articles/0")
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({ msg: "Bad request" });
+      });
+  });
```
Comment (Dutton1990) [inline __tests__/endpoint.test.js:88]: This should be a 404 because the data type of article id is correct - number. 

I wouldn't use 0 as an example though, use an impossibly big number like 5000 or something.

A 400 would be for something like /api/articles/hello - wrong data type of string. You should have a test for both!

File: news.controllers.js [hunk 2]
```diff
@@ -21,7 +21,26 @@ function getEndpoints(request, response, next) {
     });
 }
 
+function getArticleById(request, response, next) {
+  const { article_id } = request.params;
+
+  if (isNaN(article_id) || article_id <= 0) {
+    return next({ status: 400, msg: "Bad request" });
+  }
```
Comment (Dutton1990) [inline news.controllers.js]: You don't need to check whether the article_id a number or not, PSQL will do that for you. 

In the model, PSQL will try to select an article by an id that isn't a number and fail. It will return it's own error. You can then catch this error in the catch block below and send it to next. 

Then you will need a error handler to send the error to. You will in this case have a code you can check on err.code and use that to send a response to the client.

File: news.controllers.js [hunk 3]
```diff
@@ -21,7 +21,26 @@ function getEndpoints(request, response, next) {
...
+  readArticleById(article_id)
+    .then((article) => {
+      if (!article) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
```
Comment (Dutton1990) [inline news.controllers.js]: The above is good but deal with this in the model instead. When you get the rows back in the .then, check the rows instead and Promise.reject there. Controller should just deal with request and response

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Just some changes I'd like please George! Close though, just a bit of tweaking. You need an extra error test and to change your current error test. Also a few suggestions to tidy this up a bit and make use of error handlers!

Once you've made these changes please push to this branch again and re-send the pr to nchelp pr, cheers!

File: app.js [hunk 4]
```diff
@@ -5,13 +5,16 @@ const {
...
-const { handleCustomErrors } = require("./error.controllers.js");
+const { handle400s, handleCustomErrors } = require("./error.controllers.js");
 
 app.get("/api/topics", getTopics);
+
 app.get("/api", getEndpoints);
+
 app.get("/api/articles/:article_id", getArticleById);
 
-app.use(handleCustomErrors);
```
Comment (Augs0) [inline app.js]: Why did you remove your custom errors here? We still need those for 404s in the database (as opposed to 404 paths in the url)

**Review by Augs0 — APPROVED** [review]
Looks like you have hit all the points Christian mentioned, well done! I've left a brief comment on you app.js file as it looks like you _might_ have removed something you still need. You're okay to merge :)
